include/vrv/object.h

```diff
@@ -804,6 +804,16 @@ class Object : public BoundingBox {
      */
     virtual int CalcArtic(FunctorParams *) { return FUNCTOR_CONTINUE; }
 
+    /**
+     * Calculate the vertical position adjustment for the beam if it overlaps with layer elements
+     */
+    virtual int AdjustBeams(FunctorParams *) { return FUNCTOR_CONTINUE; }
+
+    /**
+     * Apply position adjustment that has been calculated previously
+     */
+    virtual int AdjustBeamsEnd(FunctorParams *) { return FUNCTOR_CONTINUE; }
+
```

Comment:
Usually I put them together (see AlignVertically above) so we straight away see there is an `*End` functor. But you did put a specific comment so it makes sense to have it like it.